app/src/main/java/com/itsaky/androidide/data/projectInfo/ProjectInfoRepository.kt

```diff
@@ -0,0 +1,33 @@
+/*
+ *  This file is part of AndroidIDE.
+ *
+ *  AndroidIDE is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  AndroidIDE is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *   along with AndroidIDE.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.itsaky.androidide.data.projectInfo
+
+import kotlinx.coroutines.flow.Flow
+import java.io.File
+
+interface ProjectInfoRepository {
+
+  fun getAllProjectInfoStream(): Flow<List<ProjectInfo>>
+
+  fun getProjectInfoStream(file: File): Flow<ProjectInfo?>
```

Comment:
This seems to be unused and as a result, `ProjectInfoDao.getProjectInfo(File)` is unused as well.